src/PrestaShopBundle/Controller/Admin/Sell/Order/CreditSlipController.php

```diff
@@ -56,20 +60,20 @@ class CreditSlipController extends FrameworkBundleAdminController
     #[AdminSecurity("is_granted('read', request.get('_legacy_controller'))")]
     public function indexAction(
         Request $request,
-        CreditSlipFilters $creditSlipFilters
+        CreditSlipFilters $creditSlipFilters,
+        #[Autowire(service: 'prestashop.core.grid.factory.credit_slip')] GridFactory $creditSlipGridFactory,
+        #[Autowire(service: 'prestashop.admin.credit_slip_options.form_handler')] Handler $creditSlipOptionsFormHandler,
     ) {
-        $creditSlipGridFactory = $this->get('prestashop.core.grid.factory.credit_slip');
         $creditSlipGrid = $creditSlipGridFactory->getGrid($creditSlipFilters);
 
-        $creditSlipOptionsFormHandler = $this->getSlipOptionsFormHandler();
         $creditSlipOptionsForm = $creditSlipOptionsFormHandler->getForm();
         $creditSlipOptionsForm->handleRequest($request);
 
         if ($creditSlipOptionsForm->isSubmitted() && $creditSlipOptionsForm->isValid()) {
             $errors = $creditSlipOptionsFormHandler->save($creditSlipOptionsForm->getData());
 
             if (empty($errors)) {
-                $this->addFlash('success', $this->trans('Update successful', 'Admin.Notifications.Success'));
+                $this->addFlash('success', $this->trans('Update successful', [], 'Admin.Notifications.Success'));
```

Comment:
Regarding the error under this one you can replace `flashErrors` with `addFlashErrors`